src/Symfony/Component/Console/Helper/WaitForTerminalInputHelper.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Symfony\Component\Console\Helper;
+
+/**
+ * WaitForTerminalInputHelper allows signals to run before a blocking user input
+ * reading call, like when using `fgets` or `fgetc` for example.
+ *
+ * Usage:
+ *
+ *     // Wait for input before all input reads
+ *     (new WaitForTerminalInputHelper($inputStream))->waitForInput();
+ *
+ *     ...read input
+ *
+ * @internal
+ */
+final class WaitForTerminalInputHelper
```

Comment:
I'd rather merge this class into the existing one, doable?

Reply:
Doable but I'll have to modify `TerminalInputHelper` since the _stty_ is only required for hidden/autocomplete/choice fields.
Its usage will be a little bit more complex (take the `shell_exec` call and signals handling out of the constructor, split the `waitForInput` method, etc.).
Do you want me to proceed?